server/src/collection.rs

```diff
@@ -11,19 +11,20 @@ use oo7::{
         api::{Properties, SecretInner},
         ServiceError,
     },
+    portal,
     portal::Keyring,
 };
 use tokio::sync::{Mutex, RwLock};
 use zbus::{interface, object_server::SignalEmitter, zvariant};
 use zvariant::{ObjectPath, OwnedObjectPath};
 
-use crate::{item, service_manager::ServiceManager};
+use crate::{error::Error, item, service_manager::ServiceManager};
 
 #[derive(Debug)]
 #[allow(unused)]
 pub struct Collection {
     // Properties
-    items: Mutex<Vec<item::Item>>,
+    items: Mutex<Vec<OwnedObjectPath>>,
```

Comment:
Why is this storing paths instead of items?

Reply:
We really don't need to store the items itself. Just the path is enough for every usecase (I have tested this). Otherwise we have have to introduce cloning everywhere.

Reply:
Not sure why you are "afraid" of Clone trait. All of involved types are Clone, no?

Reply:
I feel "not cloning the secret service interface types" approach more natural. Because, we create an object and then we dispatch it. When zbus takes care of every scenario nicely, why keep a clone just to access its attributes.

Reply:
There is probably a mis-understanding from your side what Clone means here. It just increments the ref count, no copy of the data or anything like that, as all the inner types are either Copy (bool/numerals) or are wrapped in Arc.

Reply:
You can keep it like this if you prefer.

Reply:
I switched it back to using property types, as get_ref/get_ref_mut gives us a global lock, so it will make debugging async code harder in this "complex" project